xclim/subset.py

```diff
@@ -125,11 +182,412 @@ def func_checker(*args, **kwargs):
                     kwargs[lon] -= 360
                 else:
                     kwargs[lon][kwargs[lon] < 0] -= 360
+
         return func(*args, **kwargs)
 
     return func_checker
 
 
+def wrap_lons_and_split_at_greenwich(func):
+    @wraps(func)
+    def func_checker(*args, **kwargs):
+        """
+        A decorator to split and reproject polygon vectors in a GeoDataFram whose values cross the Greenwich Meridian.
+         Begins by examining whether the geometry bounds the supplied cross longitude = 0 and if so, proceeds to split
+         the polygons at the meridian into new polygons and erase a small buffer to prevent invalid geometries when
+         transforming the lons from WGS84 to WGS84 +lon_wrap=180 (longitudes from 0 to 360).
+
+         Returns a GeoDataFrame with the new features in a wrap_lon WGS84 projection if needed.
+        """
+        try:
+            poly = kwargs["poly"]
+            x_dim = kwargs["x_dim"]
+            wrap_lons = kwargs["wrap_lons"]
+        except KeyError:
+            return func(*args, **kwargs)
```

Comment:
It seems dangerous to me to let a decorator "fail" silently. I think if you use a decorator in your code, you expect it to work, no ?

Reply:
I think this is in case the function that gets called doesn't require any of this treatment. I was hoping to show that it could leave early if the kwargs aren't present, but I could raise an error or a warning to suggest that this check isn't being performed?